lepton/starterkits/FLUX_NIMs_Interactive_Lab.ipynb

```diff
@@ -0,0 +1,2674 @@
+    ")\n",
+    "\n",
+    "setup_button.on_click(setup_lepton_cli)\n",
+    "display(setup_button)\n",
+    "display(setup_output)\n",
+    "\n",
+    "print(\"\\n💡 Where to get credentials:\")\n",
+    "print(\"  • NGC API Key: https://ngc.nvidia.com/ → Generate API Key\")\n",
+    "print(\"\\n📋 What this does:\")\n",
+    "print(\"  • Installs leptonai package with: pip install --upgrade --force-reinstall --no-cache-dir leptonai\")\n",
+    "print(\"  • Uninstalls old versions first to ensure clean installation\")\n",
+    "print(\"  • Finds working 'lep' command (official CLI)\")\n",
+    "print(\"  • Provides manual login instructions\")\n",
+    "print(\"  • Configures NGC API key for NIM deployment\")\n",
+    "print(\"\\n⚠️  Note: Manual login required - automated authentication has limitations\")\n",
+    "print(\"\\n🔄 Installation Process:\")\n",
+    "print(\"  • Uninstalls any existing leptonai package\")\n",
+    "print(\"  • Installs latest version with force reinstall\")\n",
+    "print(\"  • Skips pip cache to ensure fresh download\")\n",
+    "print(\"  • Verifies installation and version\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2.2: Deploy FLUX NIM Endpoints\n",
+    "\n",
+    "### Automated Endpoint Deployment\n",
+    "\n",
+    "Deploy FLUX NIMs as Lepton endpoints directly from the notebook. This creates production-ready API endpoints you can use immediately."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# FLUX NIM Endpoint Deployment using Lepton CLI - CORRECTED VERSION\n",
+    "# Expected time: 3-5 minutes per model (deployment + readiness check)\n",
+    "# Resource requirements: 1 GPU (A10/A100) per endpoint\n",
+    "\n",
+    "# Deployment configuration widgets\n",
+    "nim_model_selector = widgets.Dropdown(\n",
+    "    options=[\n",
+    "        ('FLUX.1 Dev - Text-to-Image Generation', 'flux.1-dev')\n",
+    "    ],\n",
+    "    value='flux.1-dev',\n",
+    "    description='NIM Model:',\n",
+    "    style={'description_width': 'initial'},\n",
+    "    layout=widgets.Layout(width='400px')\n",
+    ")\n",
+    "\n",
+    "endpoint_name_widget = widgets.Text(\n",
+    "    value='',\n",
+    "    placeholder='my-flux-nim (auto-generated if empty)',\n",
+    "    description='Endpoint Name:',\n",
+    "    style={'description_width': 'initial'},\n",
+    "    layout=widgets.Layout(width='400px')\n",
+    ")\n",
+    "\n",
+    "# Resource configuration with timing estimates\n",
+    "resource_type_widget = widgets.Dropdown(\n",
+    "    options=[\n",
+    "        ('H200 GPU (Your DevPod) - Fastest deployment, ~2-3 min', 'gpu.1xh200'),\n",
+    "        ('Small - A10 GPU (faster deployment, ~3-4 min)', 'gpu.a10'),\n",
+    "        ('Medium - A100-40GB GPU (~4-5 min)', 'gpu.a100-40gb'),\n",
+    "        ('Large - A100-80GB GPU (~5-6 min)', 'gpu.a100-80gb')\n",
```

Comment:
These values likely aren't universal. We probably want to have a user-input field for resource shape, or you can query the resource shapes with `lep node resource-shape` which will return a table of available shapes that you can pre-populate a drop-down with. The blank user-input field is probably easiest for now.